src/ItemsContext.js

```diff
@@ -4,20 +4,21 @@ import firebase from 'firebase';
 export const ItemsContext = createContext();
 
 const ItemsProvider = props => {
-  //set useState([]) for list
   const [items, setItems] = useState([]);
   const db = firebase.firestore();
   const userToken = localStorage.getItem('token');
 
   useEffect(() => {
     db.collection('items')
       .where('token', '==', userToken)
-      .get()
-      .then(snapshot => {
-        const list = snapshot.docs.map(doc => doc.data());
+      .onSnapshot(querySnapshot => {
+        const list = querySnapshot.docs.map(doc => {
+          const newObj = { ...doc.data(), id: doc.id };
+          return newObj;
+        });
```

Comment:
Can you explain what's exactly what we're doing here? If I'm reading this correctly, I think we both came up with different solutions for the same problem!

Reply:
Sure! Whenever we changed the checkbox state and updated the last purchased time field, we noticed that the database reflected the update but not the items array returned by the Itemscontext. In the firebase docs, there's a method called onSnapshot that actually listens to changes and updates the data. Once we updated the code to use onSnapshot instead of get, the changes were updating right away without refreshing the page.

On Line 16 - we created a new object that contained all the items returned by doc.data() plus an id property which is the document id that we see in the firebase database. By assigning an id to every item on the shopping list, it is very easy easy to use the doc method (provided by firebase) and say: update item with id:123 to have a last purchase date of blah blah. :star:

Reply:
Awesome! Thank you so much for explaining it. I figured that's basically what `onSnapshot` was doing, but I just wanted to double check. Great job! 👏

Reply:
Alright! I'm glad you found this elegant solution in the Firebase documentation.